### PR TITLE
Long scan timeout for #267

### PR DESCRIPTION
**Pull request by pombredanne**
In #267 the key issue is some pathological, hard to diagnose scan that takes a long time and eventually uses a lot of RAM to get prockilled  (at least on Linux).

This should both help handle the issue and help diagnose it:

* a scan will be interrupted if it takes too long and exceeds a timeout: so if there is a problem somewhere, it should timeout rather than suck all the memory and get proc killed. The approach is to run each scan in a thread with a timeout using code from @dano
The timeout is computed for each file being scanned. It has a minimum of 60 seconds and a maximum of 600 seconds.  Each file bigger than 1MB gets an extra 30 of timeout for each extra MB up to the max  600s.

* a scan can run on multiple processes with the `-n or --processes` option (this is a nice side effect of handling timeouts and can also speed up scanning of course, but would use more RAM). When running on multiple process the memory used by each process may vary but would be about 600MB + size of the scanned file in memory + some extras.

* -`-verbose` output is always send to stdout, as you requested by @sschuberth which will help to diagnose issues for background, CI-like scan jobs. So running a --verbose scan on this would likely tell exactly which is the file that is causing an error.

* a summary of the scan statistics is displayed at the end of a scan: this helps diagnose errors as it does list all the paths that failed to scan alright or did timeout. It also provide some timing and scanning speed details.

### CODE REVIEW

**Comment by pombredanne** [comment]
The latest commits now make the memory check the default

File: src/scancode/cli.py [hunk 1]
```diff
@@ -262,102 +300,201 @@ def scancode(ctx, input, output_file, copyright, license, package,
...
+            _progress_line = verbose and _scanned_path or fileutils.file_name(_scanned_path)
+            return style('Scanning: ') + style(_progress_line, fg=_scan_success and 'green' or 'red')
+
+    scanning_errors = []
+    files_count = 0
+    with utils.progressmanager(scanned_files, item_show_func=scan_event, show_pos=True, verbose=verbose, quiet=quiet) as scanned:
+        while True:
+            try:
+                result = scanned.next()
+                scan_sucess, scanned_rel_path = result
```
Comment (sschuberth) [inline src/scancode/cli.py]: Minor typo: Should say "scan_success".

**Comment by sschuberth** [comment]
Looks good to me from a cursory view, thanks for working on this. On the CI side, the one Travis build seems to have timed out and probably just needs to be retriggered. For AppVeyor things look a bit worse, probably some tweaking of `TEST_MAX_MEMORY` is required?

**Comment by pombredanne** [comment]
@sschuberth I fixed the typos and a few more. Thanks ++ for this review! I need to fix the CI failures now.

**Comment by pombr** [comment]
And at least one of the builds passed on Mac too: https://travis-ci.org/nexB/scancode-toolkit/builds/176613254

**Comment by pombr** [comment]
Linux and Windows seem OK

**Comment by pombr** [comment]
All tests pass now: :green_apple: :green_heart:

**Comment by sschuberth** [comment]
Let's get it merged for some wider exposure! :-)

**Comment by pombredanne** [comment]
@sschuberth agreed. Further fixes if needed can branch as needed.

**Pull request merged**